### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by CRR257**

### CODE REVIEW

File: staff/clara-rufi/Precurs2019/PasapalabraTimer/Pasapalabra.js [hunk 1]
```diff
@@ -0,0 +1,413 @@
...
+        }
+    }, 1000);
+}
+
+
+
+window.onload = function () {
+    var twoMinutes = 3 * 1,
+        display = document.querySelector('#time');
+    startTimer(twoMinutes, display);
```
Comment (robert-z) [inline staff/clara-rufi/Precurs2019/PasapalabraTimer/Pasapalabra.js]: You are calling `startTimer()` function in an `onload` event. This behavior start the timer a few seconds after to enter in the website.

I think you should call `startTimer()` function when you press the **start button**.

**Comment by carlosclatg** [comment]
In general, for bingo.js
Indent code correctly, sometimes difficult to read.
For each column same function, may gather into one function.
Ok, the other functions.

File: staff/clara-rufi/Precurs2019/BingoHTML/Bingo2.js [hunk 2]
```diff
@@ -0,0 +1,373 @@
+document.getElementById('cartro').style="display:none;"
```
Comment (gcelma) [inline staff/clara-rufi/Precurs2019/BingoHTML/Bingo2.js]: Està molt guay, lo únic que revisaria és que quan ja han sortit tots els números del cartró et deixa seguir prement seguent i segueixen sortint números.

**Comment by carlosclatg** [comment]
Pasapalabra:
pasapalabra.html: input welcome, is available to write when accessing. It may only be used to display not towrite.
pasalabra.js: No time for play, timer ends up fast.

File: staff/clara-rufi/Precurs2019/PasapalabraTimer/Pasapalabra.js [hunk 3]
```diff
@@ -0,0 +1,413 @@
...
+    document.getElementById("tiempo").style="display:block;"
+    document.getElementById("minuts").style="display:block;"
+    timer();
+    document.getElementById("showQuestion").value = questions[n].question;
+}
+
+// timer 
+
+function timer(){
+    startTimer()
```
Comment (robert-z) [inline staff/clara-rufi/Precurs2019/PasapalabraTimer/Pasapalabra.js]: You need to pass two variable.

```javascript
function startTimer(duration, display) {
...
}
```

File: staff/clara-rufi/Precurs2019/Calculadora_gat/Cookies_calculator.html [hunk 4]
```diff
@@ -0,0 +1,30 @@
+<!DOCTYPE html>
```
Comment (gcelma) [inline staff/clara-rufi/Precurs2019/Calculadora_gat/Cookies_calculator.html]: Toto OK!

File: staff/clara-rufi/Precurs2019/Calculadora_gat/app/app.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
+function calculate(){
```
Comment (gcelma) [inline staff/clara-rufi/Precurs2019/Calculadora_gat/app/app.js]: Tot OK!

File: staff/clara-rufi/Precurs2019/Calculadora_gat/styles/gat.css [hunk 6]
```diff
@@ -0,0 +1,131 @@
+*{
```
Comment (gcelma) [inline staff/clara-rufi/Precurs2019/Calculadora_gat/styles/gat.css]: Tot OK!

File: staff/clara-rufi/Precurs2019/PasapalabraTimer/Pasapalabra.js [hunk 7]
```diff
@@ -0,0 +1,413 @@
...
+
+function login(){
+
+var name = document.getElementById("nameUser").value
+document.getElementById("welcome").value = ""
+    if (name.length >=2){
+        nameUser.push(name)
+        console.log(nameUser)
+        document.getElementById("welcome").value += "Bienvenido " + nameUser + " !!"
+        document.getElementById('start').style="display:block;"
```
Comment (quinwacca) [inline staff/clara-rufi/Precurs2019/PasapalabraTimer/Pasapalabra.js]: Ya que el botón de login deja de ser útil si ya has introducido un nombre, y para evitar añadir nombres infinitos... prueba con : 
`document.getElementById("login").disabled = true;` ( o algo así ).

Ah ! Y ojo con los punto y coma!

File: staff/clara-rufi/Precurs2019/PasapalabraTimer/Pasapalabra.js [hunk 8]
```diff
@@ -0,0 +1,413 @@
...
+    document.getElementById("showQuestion").value = questions[n].question;
+}
+
+// timer 
+
+function timer(){
+    startTimer()
+}
+
+function startTimer(duration, display) {
```
Comment (robert-z) [inline staff/clara-rufi/Precurs2019/PasapalabraTimer/Pasapalabra.js]: In order to the countdown works you should do some changes in `startTimer()` function. I show you an example:

```javascript
function startTimer(duration, display) {
    
    var __timer = setInterval(function () {
                
        minutes = Math.floor(duration / 60);
        seconds = duration-Math.round(minutes *60);

        minutes = minutes < 10 ? "0" + minutes : minutes;
        seconds = seconds < 10 ? "0" + seconds : seconds;

        display.textContent = minutes + ":" + seconds;

        if (duration <= 0) {
            clearInterval(__timer)
            document.getElementById("timeFinished").style="display:block;"
            document.getElementById("timeFinished").value = "Tiempo acabado"
            document.getElementById("time").style="display:none;"
            document.getElementById("tiempo").style="display:none;"
            document.getElementById("minuts").style="display:none;"
            document.getElementById('letters').style="display:none;"
            document.getElementById('showQuestion').style="display:none;"
            document.getElementById('userAnswer').style="display:none;"
            document.getElementById('acceptar').style="display:none;"
            document.getElementsByClassName('letter').style="display:none;"
            document.getElementById('Okrepeat').style="display:none;"//pq no es mostri
            document.getElementById('repeatPasapalabra').style="display:none;"
            document.getElementById('showresults').style="display:none;"
            document.getElementById('results').style="display:none;"
            document.getElementById("end").style="display:none;" 
            document.getElementById("questionPlayAgain").style="display:block;"
            document.getElementById("playAgainYes").style="display:block;"
            document.getElementById("playAgainNo").style="display:block;"
           
        }
        duration--;
    }, 1000);
}
```

If you want tomorrow we can speak about this. Before doing the change you should also change all `startTimer` calls. I will try to explain you this in the following comments.

File: staff/clara-rufi/Precurs2019/PasapalabraTimer/Pasapalabra.js [hunk 9]
```diff
@@ -0,0 +1,413 @@
...
+    document.getElementById("results").style="display:none;"
+    document.getElementById("timeFinished").style="display:none;"
+    document.getElementById("time").style="display:block;"
+    document.getElementById("tiempo").style="display:block;"
+    document.getElementById("minuts").style="display:block;"
+    /*document.getElementById("userAnswer").value = "";
+    document.getElementById("showQuestion").value = "";*/
+
+
+    timer();
```
Comment (robert-z) [inline staff/clara-rufi/Precurs2019/PasapalabraTimer/Pasapalabra.js]: In referenece to this [comment](https://github.com/manuelbarzi/skylab-bootcamp-201901/pull/42/files#r246115218), in this case you should call `startTimer` function.

```javascript
startTimer(180, document.querySelector('#time'));
```

As you can see, I'm pass the total seconds (180 = 3 minuts) and the querySelector.

File: staff/clara-rufi/Precurs2019/PasapalabraTimer/Pasapalabra.js [hunk 10]
```diff
@@ -0,0 +1,413 @@
...
+    
+    
+    window.onload = function () {
```
Comment (robert-z) [inline staff/clara-rufi/Precurs2019/PasapalabraTimer/Pasapalabra.js]: This `onload` event is not necessary. You are calling the `startTimer` function before.

File: staff/clara-rufi/Precurs2019/PasapalabraTimer/Pasapalabra.js [hunk 11]
```diff
@@ -0,0 +1,413 @@
...
+    document.getElementById('letters').style="display:block;"
+    document.getElementById('showQuestion').style="display:block;"
+    document.getElementById('userAnswer').style="display:block;"
+    document.getElementById('loginUser').style="display:none;" 
+    document.getElementById('acceptar').style="display:block;"
+    document.getElementById('end').style="display:block;"
+    document.getElementById("time").style="display:block;"
+    document.getElementById("tiempo").style="display:block;"
+    document.getElementById("minuts").style="display:block;"
+    timer();
```
Comment (robert-z) [inline staff/clara-rufi/Precurs2019/PasapalabraTimer/Pasapalabra.js]: In referenece to this [comment](https://github.com/manuelbarzi/skylab-bootcamp-201901/pull/42/files#r246115218), in this case you should call startTimer function.

```javascript
startTimer(180, document.querySelector('#time'));
```

As you can see, I'm pass the total seconds (180 = 3 minuts) and the querySelector.

File: staff/clara-rufi/Precurs2019/PasapalabraTimer/Pasapalabra.js (hunk 1 above)
Comment (CRR257) [reply staff/clara-rufi/Precurs2019/PasapalabraTimer/Pasapalabra.js]: thanks, I will try it!

File: staff/clara-rufi/Precurs2019/BingoHTML/Bingo2.js (hunk 2 above)
Comment (CRR257) [reply staff/clara-rufi/Precurs2019/BingoHTML/Bingo2.js]: Si...del bingo no està acabat al 100% :), gracies x la suggerència

File: staff/clara-rufi/Precurs2019/BingoHTML/Bingo.html [hunk 12]
```diff
@@ -0,0 +1,94 @@
...
+    <link rel = "stylesheet" href="Styles/Bingo.css">
+    <title>Bingo</title>
+</head>
+<body>
+
+    <div id = "welcome">
+    <h2>Bienvenido al Bingo!</h2>
+    <div id = "requestName">Escribe aqui tu nombre de usuario: </div>
+    <input type ="text" id= "nameUser" maxlength="15">
+    <button onclick= "login(); play()" id = "play">Jugar!</button>
```
[the remark this answers is not in the document]
Comment (CRR257) [reply staff/clara-rufi/Precurs2019/BingoHTML/Bingo.html]: ok, I will try it, thanks ;)

File: staff/clara-rufi/Precurs2019/PasapalabraTimer/Pasapalabra.js (hunk 7 above)
Comment (CRR257) [reply staff/clara-rufi/Precurs2019/PasapalabraTimer/Pasapalabra.js]: ok, provaré, gracias! :)

File: staff/clara-rufi/Precurs2019/PasapalabraTimer/Pasapalabra.js (hunk 8 above)
Comment (CRR257) [reply staff/clara-rufi/Precurs2019/PasapalabraTimer/Pasapalabra.js]: ok thanks. in fact, i found another countdown and i wanted to try it. but thanks for your suggestions, i will consider them, too :)

**Pull request merged**